### PR TITLE
doc: add file name details 5088 v2 

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5088

Describe changes:
- updated signature examples to use rule container
- added asterisk to denote sticky buffer for http keyword table
- HTTP2 does not currently support file name keyword usage so that was left out of docs

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: doc/userguide/rules/http-keywords.rst [hunk 1]
```diff
@@ -843,4 +843,16 @@ Notes
...
+
+file.name
+---------
+
+The ``file.name`` keyword can be used at the HTTP application level.
+
+Example::
+
+  alert http any any -> any any (msg:"http layer file.name keyword usage"; \
+  file.name; content:"picture.jpg"; classtype:bad-unknown; sid:1; rev:1;)
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:856]: Any special reason not to use the `example-rule` container here?

**Review by jufajardini — APPROVED** [review]
It looks good to me, thanks! :)  Left an inline question, but nothing that prevents merging, from my end.

File: doc/userguide/rules/http-keywords.rst (hunk 1 above)
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:856]: Hm, it looks like I missed updating that one. I can either submit a new PR or my plan is to have a larger update to http-keywords.rst based on our conversation around the keyword table and other updates around https://redmine.openinfosecfoundation.org/issues/3025 in the next couple days

Comment (jufajardini) [reply doc/userguide/rules/http-keywords.rst:856]: that second option sounds good to me :)

**Comment by victorjulien** [comment]
Merged in #9985, thanks!

**Pull request closed without merging**